components/measurement/FeedbackBox.js

```diff
@@ -0,0 +1,254 @@
+import React, { useEffect, useMemo, useState } from 'react'
+import { Controller, useForm } from 'react-hook-form'
+import { FormattedMessage, useIntl } from 'react-intl'
+import { Box, Button, Link, Text, theme } from 'ooni-components'
+import { RadioGroup, RadioButton } from 'components/search/Radio'
+import useStateMachine from '@cassiozen/usestatemachine'
+import SpinLoader from 'components/vendor/SpinLoader'
+import { submitFeedback, getAPI } from 'lib/api'
+
+const okValues = ['ok', 'ok.unreachable', 'ok.broken', 'ok.parked']
+const blockedValues = [
+  { top: 'blocked' },
+  { 
+    top: 'blocked.blockpage',
+    sub: [
+      'blocked.blockpage.http',
+      'blocked.blockpage.dns',
+      'blocked.blockpage.server_side',
+      'blocked.blockpage.server_side.captcha'
+    ]
+  },
+  { top: 'blocked.dns',
+    sub: ['blocked.dns.inconsistent', 'blocked.dns.nxdomain']
+  }
+]
+
+const QuestionText = ({i18nKey}) => (
+  <Text fontSize={16} mb={3}><FormattedMessage id={i18nKey} /></Text>
+)
+
+const FeedbackBox = ({user, report_id, setShowModal, previousFeedback}) => {
+  const intl = useIntl()
+  const [error, setError] = useState(null)
+
+  const { handleSubmit, control, watch, setValue, getValues } = useForm()
+
+  const [state, send] = useStateMachine({
+    initial: 'initial',
+    states: {
+      initial: {
+        on: { 
+          CLOSE: 'closed',
+          IS_BLOCKING: 'isBlocking'
+        },
+      },
+      isBlocking: {
+        on: {
+          NOT_BLOCKING: 'notBlocking',
+          BLOCKING: 'blocking',
+        }
+      },
+      blocking: {
+        on: {
+          CANCEL: 'initial',
+          SUBMIT: 'submit'
+        }
+      },
+      notBlocking: {
+        on: {
+          CANCEL: 'initial',
+          SUBMIT: 'submit'
+        }
+      },
+      closed: {},
+      submit: {
+        effect({ send, setContext, event, context }) {
+          const { nested, feedback } = getValues()
+          const feedbackParams = {
+            measurement_uid: report_id,
+            status: nested || feedback
+          }
+          submitFeedback(feedbackParams)
+            .then(() => send('SUCCESS'))
+            .catch(({response}) => {
+              setError(response?.data?.error || 'unknown')
+              send('FAILURE')
```

Comment:
There is something odd with how the failures are being handled. I tried from the test instance to submit feedback, but the UI showed a successful submission, while according to the network inspector there was actually a CORS error.